### ISSUE TITLE
[dif_kmac] Functions for SCA

### ISSUE THREAD

**Issue by alphan** [lowRISC/opentitan#7842]
#7757 adds an initial implementation for SHA3/KMAC SCA program `sha3_serial.c`. I had to perform several writes to KMAC registers to unblock that PR but most of those should be replaced when they are implemented in `dif_kmac` library:
- [ ] `kmac_block_until_idle()`: We can remove this if we implement a function like `dif_kmac_get_status()` that returns the status bits in the `KMAC_STATUS` register.
- [ ] `kmac_reset()`: We can remove this when `dif_kmac_reset()` is implemented (currently only declared in `dif_kmac.h`)
- [x] `kmac_absorb_end()`: I think we can wait until we finalize the way we capture traces (see #7841).
- [ ] Move/merge SCA mode software interface functions into KMAC DIF for easier maintenance (see https://github.com/lowRISC/opentitan/pull/8240#discussion_r711417724)

**Comment by vogelpi** [comment]
I've done a PR here #8003 to facilitate SCA measurements. With these changes, software can exercise KMAC using the following sequence:
1. Configure mode and key etc.
2. Write data to the message FIFO.
3. Append output length to message FIFO.
4. Send START and PROCESS commands, got to sleep and wait for `event_absorbed` interrupt. After a configurable delay. KMAC will first do the absorb and then the squeeze.
5. Read the digest to verify correct operation.
6. Start over.

I've implemented a simple self-checking test to verify this on my branch here: https://github.com/vogelpi/opentitan/commits/kmac-sca-dev

**Comment by vogelpi** [comment]
Update: #8240 changes the way how we capture traces. `sha3_serial` now uses the FPGA SCA mode of KMAC. As for AES, Ibex configures the timer, starts the processing (with a delay), and then enters sleep (`wfi`).

**Comment by etterli** [comment]
@vogelpi Is this still relevant?

**Comment by vogelpi** [comment]
> [@vogelpi](https://github.com/vogelpi) Is this still relevant?

No, lets close this. How we do FPGA SCA for KMAC is now quite different anyway.
